en/middleware/micrortps.md

```diff
@@ -0,0 +1,271 @@
+For example, the following functions would be generated for the *sensor_combined.msg*:
+
+```sh
+void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+```
+
+
+## Client (PX4 Firmware)
+
+The *Client* source code is generated, and the executable is automatically built and included when you make the firmware (in the normal way). For example, to build and upload the firmware:
+
+For NuttX/Pixhawk flight controllers:
+```sh
+make px4fmu-v4_default upload
+```
+For Qualcomm Snapdragon Flight:
+```sh
+$ make eagle_default upload
+  ```
+
+After uploading the firmware, the *Client* application can be launched by typing its name and passing a variable number of arguments as shown below:
+
+```sh
+> micrortps_client start|stop [options]
+  -t <transport>          [UART|UDP] Default UART
+  -d <device>             UART device. Default /dev/ttyACM0
+  -u <update_time_ms>     Time in ms for uORB subscribed topics update. Default 0
+  -l <loops>              How many iterations will this program have. -1 for infinite. Default 10000
+  -w <sleep_time_ms>      Time in ms for which each iteration sleep. Default 1ms
+  -b <baudrate>           UART device baudrate. Default 460800
+  -p <poll_ms>            Time in ms to poll over UART. Default 1ms
+  -r <reception port>     UDP port for receiving. Default 2019
+  -s <sending port>       UDP port for sending. Default 2020
+```
+
+> **Note** If the selected UART port is busy, it's possible that the MAVLink application is already being used. If it is the case, you can stop MAVLink from NuttShell by typing:
+  ```sh
+  > mavlink stop-all
+  ```
+
+
+## Agent (Off Board FastRTPS Interface)
+
+The *Agent* code is automatically generated when you compile/build the associated PX4 firmware (into **build_BUILDPLATFORM/src/modules/micrortps_bridge/micrortps_agent/**)
+
+To build the *Agent* application, compile the code:
+```sh
+cd src/modules/micrortps_bridge/microRTPS_agent
+mkdir build && cd build
+cmake ..
+make
+  ```
+
+> **Note** To cross-compile for the *Qualcomm Snapdragon Flight* platform see [this link](https://github.com/eProsima/PX4-FastRTPS-PoC-Snapdragon-UDP#how-to-use).
+
+To launch the *Agent* run:
+
+```text
+$ ./micrortps_agent [options]
+  -t <transport>          [UART|UDP] Default UART
+  -d <device>             UART device. Default /dev/ttyACM0
+  -w <sleep_time_us>      Time in us for which each iteration sleep. Default 1ms
+  -b <baudrate>           UART device baudrate. Default 460800
+  -p <poll_ms>            Time in ms to poll over UART. Default 1ms
+  -r <reception port>     UDP port for receiving. Default 2019
+  -s <sending port>       UDP port for sending. Default 2020
+```
+
+
+## Creating a Listener application
```

Comment:
This is great. Possible to get a publisher example as well that does something realistic?